core/src/main/java/io/grpc/MethodDescriptor.java

```diff
@@ -103,4 +179,48 @@ public ResponseT parseResponse(InputStream input) {
   public InputStream streamRequest(RequestT requestMessage) {
     return requestMarshaller.stream(requestMessage);
   }
+
+  /**
+   * Parse an incoming request message.
+   *
+   * @param input the serialized message as a byte stream.
+   * @return a parsed instance of the message.
+   */
+  public RequestT parseRequest(InputStream input) {
+    return requestMarshaller.parse(input);
+  }
+
+  /**
+   * Serialize an outgoing response message.
+   *
+   * @param response the response message to serialize.
+   * @return the serialized message as a byte stream.
+   */
+  public InputStream streamResponse(ResponseT response) {
+    return responseMarshaller.stream(response);
+  }
+
+  /**
+   * Generate the fully qualified method name.
+   *
+   * @param fullServiceName the fully qualified service name that is prefixed with the package name
+   * @param methodName the short method name
+   */
+  public static String generateFullMethodName(String fullServiceName, String methodName) {
+    return fullServiceName + "/" + methodName;
+  }
+
+  /**
+   * Extract the fully qualified service name out of a fully qualified method name. May return
+   * {@code null} if the input is malformed, but you cannot rely on it for the validity of the
+   * input.
+   */
+  @Nullable
+  public static String extractFullServiceName(String fullMethodName) {
```

Comment:
No we can't. In MutableHandlerRegistryImpl.lookupMethod() it is used to get the full service name with just the full method name. We don't have a MethodDescriptor at this point.